app/post_metadata.go

```diff
@@ -422,12 +422,6 @@ func (a *App) getLinkMetadata(requestURL string, timestamp int64, isNewPost bool
 
 		client := a.HTTPService().MakeClient(false)
 		client.Timeout = time.Duration(*a.Config().ExperimentalSettings.LinkMetadataTimeoutMilliseconds) * time.Millisecond
-		mmTransport := a.HTTPService().MakeTransport(false)
-		client.Transport = mmTransport.Transport
-
-		if strings.HasPrefix(requestURL, "https://twitter.com/") || strings.HasPrefix(requestURL, "https://mobile.twitter.com/") {
-			request.Header.Add("User-Agent", "facebookexternalhit/1.1 (+http://www.facebook.com/externalhit_uatext.php)")
-		}
```

Comment:
That code was added as a workaround for twitter links as twitter doesn't send open graph data expect for certain user-agents. Why was it removed?

Reply:
Because it's actually the other way around; Twitter sends the appropriate OpenGraph data for most user-agents, specifically blocking a few that they have determined are abusive (including `Go-http-client/1.1`). Furthermore, I have checked and `Mattermost-Bot/1.1` works properly on Twitter:
```shellsession
$ # Golang default user-agent
$ curl -s -H 'User-Agent: Go-http-client/1.1' https://twitter.com/Mattermost/status/1372270097012826114 | grep -o '<meta[^>]*property="og:[^>]*>'
<meta property="og:site_name" content="Twitter" />
$ # Current link preview user-agent
$ curl -s -H 'User-Agent: facebookexternalhit/1.1 (+http://www.facebook.com/externalhit_uatext.php)' https://twitter.com/Mattermost/status/1372270097012826114 | grep -o '<meta[^>]*property="og:[^>]*>'
<meta  property="og:type" content="article">
<meta  property="og:url" content="https://twitter.com/Mattermost/status/1372270097012826114">
<meta  property="og:title" content="Mattermost on Twitter">
<meta  property="og:image" content="https://pbs.twimg.com/media/EwtIWpqWgAEvGS6.jpg:large">
<meta  property="og:image:user_generated" content="true">
<meta  property="og:description" content="“NEW: Mattermost v5.33 is now available - https://t.co/AQICWKhpx0”">
<meta  property="og:site_name" content="Twitter">
$ # New custom user-agent
$ curl -s -H 'User-Agent: Mattermost-Bot/1.0' https://twitter.com/Mattermost/status/1372270097012826114 | grep -o '<meta[^>]*property="og:[^>]*>'
<meta  property="og:type" content="article">
<meta  property="og:url" content="https://twitter.com/Mattermost/status/1372270097012826114">
<meta  property="og:title" content="Mattermost on Twitter">
<meta  property="og:image" content="https://pbs.twimg.com/media/EwtIWpqWgAEvGS6.jpg:large">
<meta  property="og:image:user_generated" content="true">
<meta  property="og:description" content="“NEW: Mattermost v5.33 is now available - https://t.co/AQICWKhpx0”">
<meta  property="og:site_name" content="Twitter">
```
Removing this special case both simplifies the code and removes a behavior that is (IMO) not the best form.

Reply:
I've tested the changes locally and it seem like that twitter links are still shown.